### PR TITLE
source(clickhouse(...)) usability

### LINKED ISSUE #16616: Poor usability of CREATE DICTIONARY query with CLICKHOUSE source.
It's not possible to avoid specifying port (server port should be implied).
It's not possible to avoid specifying user (default user should be assumed).
It's not possible to avoid specifying host (localhost should be assumed).

Documentation states that I can specify `LAYOUT(MAX 30)` and omit `MIN`.
https://clickhouse.tech/docs/en/sql-reference/statements/create/dictionary/
It does not work.

### PR DESCRIPTION
**Pull request by vdimir**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Set default `host` and `port` parameters for `SOURCE(CLICKHOUSE(...))` to current instance and set default `user` value to `'default'`.

Closes #16616

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
Maybe also use empty string as default password?

File: src/Dictionaries/ClickHouseDictionarySource.cpp [hunk 1]
```diff
@@ -17,6 +17,14 @@
 
 namespace DB
 {
+namespace
+{
```
Comment (alexey-milovidov) [inline src/Dictionaries/ClickHouseDictionarySource.cpp:21]: Either anon. namespace or static. Both of them looks redundant...

**Comment by vdimir** [comment]
> Maybe also use empty string as default password?

It is already set 
https://github.com/ClickHouse/ClickHouse/blob/66fc54633134ae8d07a93ce265f159aeeda1d436/src/Dictionaries/ClickHouseDictionarySource.cpp#L74

File: tests/queries/0_stateless/01018_ddl_dictionaries_select.sql [hunk 2]
```diff
@@ -51,6 +51,27 @@ DROP DICTIONARY database_for_dict.dict1;
...
+
+CREATE DICTIONARY database_for_dict.dict1
+(
+  key_column UInt64 DEFAULT 0,
+  second_column UInt8 DEFAULT 1,
+  third_column String DEFAULT 'qqq',
+  fourth_column Float64 DEFAULT 42.0
+)
+PRIMARY KEY key_column
+SOURCE(CLICKHOUSE(TABLE 'table_for_dict' PASSWORD '' DB 'database_for_dict'))
```
Comment (alexey-milovidov) [inline tests/queries/0_stateless/01018_ddl_dictionaries_select.sql:65]: Ok. I see that default password option is not covered by the test.

**Comment by alexey-milovidov** [comment]
Let's merge with master after #17044.
There were broken tests by the random inference of the new feature.

Also "Timeout" is due to the random failures of network connectivity in Sandbox due to interaction of Docker, iptables and HBF...

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**